src/hooks/useFetchListCallback.ts

```diff
@@ -6,26 +6,21 @@ import resolveENSContentHash from 'lib/utils/resolveENSContentHash'
 import { useCallback } from 'react'
 import { useAppDispatch } from 'state/hooks'
 
-import { getNetworkLibrary } from '../connectors'
 import { fetchTokenList } from '../state/lists/actions'
 
 export function useFetchListCallback(): (listUrl: string, sendDispatch?: boolean) => Promise<TokenList> {
-  const { chainId, library } = useActiveWeb3React()
+  const { provider } = useActiveWeb3React()
   const dispatch = useAppDispatch()
 
   const ensResolver = useCallback(
     async (ensName: string) => {
-      if (!library || chainId !== 1) {
-        const networkLibrary = getNetworkLibrary()
-        const network = await networkLibrary.getNetwork()
-        if (networkLibrary && network.chainId === 1) {
-          return resolveENSContentHash(ensName, networkLibrary)
-        }
-        throw new Error('Could not construct mainnet ENS resolver')
+      if (provider) {
```

Comment:
need to fix this

Reply:
i can't really do that with the way web3-react works. it used to work with v6 though... i'd like to implement caching the ENS and balance though

Reply:
Can't you have a separate web3-react network connector that only connects to mainnet? It doesn't need to go through the Web3ReactProvider.

Reply:
oh i see. yeah let me try that out.